src/Egret.Cli/Serialization/Xml/XmlSerializer.cs

```diff
@@ -0,0 +1,85 @@
+﻿namespace Egret.Cli.Serialization.Xml
+{
+    using System;
+    using System.IO;
+    using System.IO.Abstractions;
+    using System.Threading.Tasks;
+    using System.Xml;
+    using System.Xml.Serialization;
+
+    public class DefaultXmlSerializer
+    {
+        public Action<object, XmlAttributeEventArgs> OnUnknownAttribute { get; set; }
+
+        public Action<object, XmlElementEventArgs> OnUnknownElement { get; set; }
+
+        public Action<object, XmlNodeEventArgs> OnUnknownNode { get; set; }
+
+        public Action<object, UnreferencedObjectEventArgs> OnUnreferencedObject { get; set; }
+        
+        public async Task<T> Deserialize<T>(IFileInfo fileInfo)
```

Comment:
The way these event bindings are done this class can not be a singleton in the DI service

..I just looked, it's not even there?

It should be added as a transient in DI service